### ISSUE TITLE
Add some project info to docs

### ISSUE THREAD

**Issue by burner1024** [openlawlibrary/pygls#305]
Hi.
I'd like to request some background info added to the docs. Specifically, about the pygls inception and development goals.
I'm considering options for developing a language server, and honestly, I'd much rather deal with Python than JS, but I'm not quite clear how's pygls compared to MS vscode-languageserver.

I assume it's not a core project for OpenLawLibrary, but rather something to support your actual project, whatever that is?

I see a list of [implemented features](https://pygls.readthedocs.io/en/latest/pages/tutorial.html#hacking-the-extension), but is there a list on unimplemented ones, maybe even a roadmap? What's missing in pygls compared to vscode-languageserver? And vice versa?

Any performance, compatiblity considerations?
Python 3.7 should still run on windows 7, I think. Any plans to drop that?

Just to be clear, I'm not trying to be imposing/demanding. I understand that open source is open source. Simply want to get a clearer picture of the project, and I think that would be helpful to other people, too.

**Comment by tombh** [comment]
This is a _really_ good point. Thank you. I'll research and respond soon.

**Comment by augb** [comment]
@tombh I'll reach out to you via email to provide some background information on the origination of the project.

**Comment by tombh** [comment]
So I've made a PR, #315, that addresses some of the questions here and more.

Pygls' inception is now addressed in a [HISTORY.md](https://github.com/openlawlibrary/pygls/blob/update-example-server/HISTORY.md) file, thanks to @augb.

There's now also an [Alternatives](https://github.com/openlawlibrary/pygls/blob/update-example-server/README.md#alternatives) section that should address the pros/cons of Pygls compared to its competitors/alternatives.

@burner1024 Those implemented features that you linked to are merely for the example VSCode extension. In actual fact, with Pygls' recent v1 release that depends on [Microsoft's official lsprotocol types](https://github.com/microsoft/lsprotocol), we should in theory both support all of LSP's current _and_ potential future features.

I'm not very familiar with Python's performance characteristics, other than it's a scripting language, so compared to Language Servers written in compiled languages, its performance is, like NodeJS, going to be relatively slower, but certainly not impractical. I don't know of any reasons to drop Python 3.7 support in the near future.
